diff --git a/es/components/Checkbox/Checkbox.js b/es/components/Checkbox/Checkbox.js
index 55fdcfcb4f91eca01fcc9a7dffc44699afcfe412..f2436d2bf03aae274964af621970c7e89b511d74 100644
--- a/es/components/Checkbox/Checkbox.js
+++ b/es/components/Checkbox/Checkbox.js
@@ -45,7 +45,7 @@ var Checkbox = /*#__PURE__*/React__default.forwardRef(function (_ref, _ref2) {
     className: "".concat(prefix, "--checkbox"),
     id: id,
     ref: function ref(el) {
-      if (el && indeterminate) {
+      if (el) {
         el.indeterminate = indeterminate;
       }
 
