diff --git a/es/components/Switch/Switch.js b/es/components/Switch/Switch.js
index 15e59c89ef364a4dffd96ac4c8eb873ceaee41aa..50f344e56f3dd54a817c4d2a4c037361afcfe220 100644
--- a/es/components/Switch/Switch.js
+++ b/es/components/Switch/Switch.js
@@ -115,7 +115,6 @@ Switch.propTypes = {
 };
 Switch.defaultProps = {
   selected: false,
-  text: 'Provide text',
   onClick: function onClick() {},
   onKeyDown: function onKeyDown() {}
 };
diff --git a/lib/components/Switch/Switch.js b/lib/components/Switch/Switch.js
index 26a0ddb51f5ac737585e7b76cdd9cc46976471f7..2302412d046d4f7f082131f1b8f22cd5159282d2 100644
--- a/lib/components/Switch/Switch.js
+++ b/lib/components/Switch/Switch.js
@@ -125,7 +125,6 @@ Switch.propTypes = {
 };
 Switch.defaultProps = {
   selected: false,
-  text: 'Provide text',
   onClick: function onClick() {},
   onKeyDown: function onKeyDown() {}
 };
