diff --git a/node_modules/@carbon/ibmdotcom-react/es/components/index.js b/node_modules/@carbon/ibmdotcom-react/es/components/index.js
index 9cfec79..cd5fce7 100644
--- a/node_modules/@carbon/ibmdotcom-react/es/components/index.js
+++ b/node_modules/@carbon/ibmdotcom-react/es/components/index.js
@@ -4,7 +4,6 @@
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
-export * from './AudioPlayer';
 export * from './ButtonGroup';
 export * from './CalloutQuote';
 export * from './CalloutWithMedia';
